contributors/design-proposals/pod-resource-management.md

```diff
@@ -1,114 +1,367 @@
+container and a BestEffort container is classified as a Burstable pod.  The
+BestEffort container is not able to consume slack resources from the sibling
+Burstable container.  It must instead compete for scarce resources at the node
+level across all containers in all QoS classes which violates the design.
+1. Ability to charge per container overhead to the pod instead of the node.
+This overhead is container runtime specific.  For example, `docker` has
+an associated `containerd-shim` process that is created for each container
+which should be charged to the pod.
+1. Ability to charge any memory usage of memory-backed volumes to the pod when
+an individual container exits instead of the node.
+
+## Enabling the unified cgroup hierarchy
+
+To enable the unified cgroup hierarchy, the operator must enable the
+`--cgroups-per-qos` flag.  Once enabled, the `kubelet` will start managing
+inner nodes of the described cgroup hierarchy.
+
+## Configuring a cgroup driver
+
+The `kubelet` will support manipulation of the cgroup hierarchy on
+the host using a cgroup driver. The driver is configured via the
+`--cgroup-driver` flag.
+
+The supported values are the following:
+
+* `cgroupfs` is the default driver that performs direct manipulation of the
+cgroup filesystem on the host in order to manage cgroup sandboxes.
+* `systemd` is an alternative driver that manages cgroup sandboxes using
+transient slices for resources that are supported by that init system.
+
+Depending on the configuration of the associated container runtime,
+operators may have to choose a particular cgroup driver to ensure
+proper system behavior.  For example, if operators use the `systemd`
+cgroup driver provided by the `docker` runtime, the `kubelet` must
+be configured to use the `systemd` cgroup driver.
+
+Implementation of either driver will delegate to the libcontainer library
+in opencontainers/runc.
+
+### Conversion of cgroupfs to systemd naming conventions
+
+Internally, the `kubelet` maintains both an abstract and a concrete name
+for its associated cgroup sandboxes.  The abstract name follows the traditional
+`cgroupfs` style syntax.  The concrete name is the name for how the cgroup
+sandbox actually appears on the host filesystem after any conversions performed
+based on the cgroup driver.
+
+If the `systemd` cgroup driver is used, the `kubelet` converts the `cgroupfs`
+style syntax into transient slices, and as a result, it must follow `systemd`
+conventions for path encoding.
+
+For example, the cgroup name `/Burstable/pod_123-456` is translated to a
+transient slice with the name `Burstable-pod_123_456.slice`.  Given how
+systemd manages the cgroup filesystem, the concrete name for the cgroup
+sandbox becomes `/Burstable.slice/Burstable-pod_123_456.slice`.
+
+## Integration with container runtimes
+
+The `kubelet` when integrating with container runtimes always provides the
+concrete cgroup filesystem name for the pod sandbox.
```

Comment:
we settled on concrete.  the shims are always provided the cgroupfs syntax as found on the host.